procedures/unit-testing-basics.ipf

```diff
@@ -307,19 +314,51 @@ static Function DisableIgorDebugger()
 
 	variable debuggerState
 
+/// Set the Igor Debugger, returns the previous state
+/// @param state		3 bits to set
+///						0x01: debugger enable
+///						0x02: debug on error
+///						0x04: debug on NVAR SVAR WAVE reference error
+static Function SetIgorDebugger(state)
+	variable state
+	
+	variable prevState, enable, debugOnError, nvarSvarWave
+
 	DebuggerOptions
-	debuggerState = V_enable
+	prevState = !!V_enable | (!!V_debugOnError) << 1 | (!!V_NVAR_SVAR_WAVE_Checking) << 2
```

Comment:
Instead of multiplying with hard-coded numbers, all multiplications are with the IUTF_DEBUG - constants. Now we should have the freedom to change the bit pattern of debugMode freely.